### PR TITLE
Edit item page

### PR DESCRIPTION
**Pull request by owencooke**
### Describe your changes

This PR accomplishes the following:

- adds the "Edit Item" page feature
- generalizes Add and Edit into a superclass
- moved Add/Edit forms into their own package
- added a `goBack` method for fragment pages to traverse the stack history in `FragmentUtils`

### Issue ticket number and link

This PR addresses: https://github.com/CMPUT301F23T08/HouseHomey/issues/14

### Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Necessary java docs are present
- [x] Necessary UML diagrams are created
- [ ] Necessary tests are written

### Screenshots (if applicable)
<div style="display: flex;">

</div>

### CODE REVIEW

**Comment by owencooke** [comment]
Two additional things to think about in this PR:
1. I figured out how to define a method in MainActivity that we can use to get the itemRef collection, instead of passing it into each Fragment. This was especially useful in avoiding  Main -> Home -> View -> Edit itemRef passing. So we could potentially refactor the existing constructors to use this new method of getting the ref
2. Pros and cons of bundling Item vs. passing it as a constructor? See below:

#### Bundle passing
```java
ViewItemFragment viewItemFragment = new ViewItemFragment();
            Bundle args = new Bundle();
            args.putSerializable("item", item);
            viewItemFragment.setArguments(args);
            navigateToFragmentPage(context, viewItemFragment);
```

#### Constructor passing
```java
navigateToFragmentPage(getContext(), new EditItemFragment(item))
```

#### Bundle recieving
```java
ofNullable(getArguments())
                .map(args -> args.getSerializable("item", Item.class))
                .ifPresent(item -> {
                    this.item = item;
                    make.setText(item.getMake());
                    model.setText(item.getModel());
                    serialNumber.setText(item.getSerialNumber());
                    cost.setText(item.getCost().toString());
                    comment.setText(item.getComment());
                });
```

#### Constructor receiving
```java
 public EditItemFragment(Item item) {
        this.item = item;
    }
```

I remember they said in the lab that Bundle passing is preferred?? But this seems to be more concise and has no drawbacks that I'm aware of... food for thought 🍔

File: app/src/main/java/com/example/househomey/ItemFormFragment.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+        data.put("cost", getInputText(R.id.add_item_cost));
+        data.put("make", getInputText(R.id.add_item_make));
+        data.put("model", getInputText(R.id.add_item_model));
+        data.put("serialNumber", getInputText(R.id.add_item_serial_number));
+        data.put("comment", getInputText(R.id.add_item_comment));
+
+        // Ensure that form data can be used to create a valid Item
+        Item item;
+        try {
+            item = new Item(itemId, data);
```
Comment (ldbonkowski) [inline app/src/main/java/com/example/househomey/ItemFormFragment.java]: How about ```return new Item(itemId, data);

**Review by ldbonkowski — APPROVED** [review]
Looks great!

**Review by MatthewNeufeld — APPROVED** [review]
Absolutely gorgeous. Nice job on everything!!

**Review by Sami-Jagirdar — COMMENTED** [review]
Great stuff!

**Pull request merged**